### PR TITLE
Add ability to create API keys with 'register executor' capability.

### PR DESCRIPTION
**Pull request by vadimberezniker**
---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/app/api_keys/api_keys.tsx [hunk 1]
```diff
@@ -390,8 +422,20 @@ export default class ApiKeysComponent extends React.Component<ApiKeysComponentPr
   }
 }
 
-function isReadOnly<T extends ApiKeyFields>(apiKey?: T | null) {
+function hasCapability<T extends ApiKeyFields>(apiKey: T | null, capability: api_key.ApiKey.Capability) {
   return apiKey?.capability
```
Comment (bduffany) [inline enterprise/app/api_keys/api_keys.tsx:426]: nit: I think this could be simplified to `return Boolean(apiKey?.capability?.some(...))`

**Review by bduffany — APPROVED** [review]

File: enterprise/app/api_keys/api_keys.tsx [hunk 2]
```diff
@@ -140,7 +141,7 @@ export default class ApiKeysComponent extends React.Component<ApiKeysComponentPr
         request: new api_key.UpdateApiKeyRequest({
           id: apiKey.id,
           label: apiKey.label,
-          capability: apiKey.capability,
+          capability: apiKey.capability.slice(0),
```
Comment (bduffany) [inline enterprise/app/api_keys/api_keys.tsx:144]: nit: slightly more idiomatic to do `[...apiKey.capability]`

File: enterprise/app/api_keys/api_keys.tsx (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/app/api_keys/api_keys.tsx]: Done.

File: enterprise/app/api_keys/api_keys.tsx (hunk 2 above)
Comment (vadimberezniker) [reply enterprise/app/api_keys/api_keys.tsx]: Done.

**Comment by vadimberezniker** [comment]
@siggisim I'll wait for your LGTM before merging

**Review by siggisim — APPROVED** [review]

**Pull request merged**